Dev

KAMKillJoy opened the pull request:
RC

KAMKillJoy committed:
init

KAMKillJoy committed:
Удаление ненужных файлов и переименование codecs.py

KAMKillJoy committed:
обновление gitignore

KAMKillJoy committed:
исправление ключа pixel_format

KAMKillJoy committed:
полная переработка меню

KAMKillJoy committed:
модуль составления команды

KAMKillJoy committed:
переработка хелперов

KAMKillJoy committed:
добавление раннеров

KAMKillJoy committed:
удаление дублирующего кода

KAMKillJoy committed:
УБран хардкод кодека

KAMKillJoy committed:
теперь output_dir задаётся в одном месте

KAMKillJoy committed:
исправление меню

KAMKillJoy committed:
корректировка стандартных значений

KAMKillJoy committed:
изменено хранение допустимых опций

KAMKillJoy committed:
убраны избыточные стандартные значения

KAMKillJoy committed:
изменена логика scale_fix

KAMKillJoy committed:
починен фильр crop

KAMKillJoy committed:
pixel переименован в pixel_format

KAMKillJoy committed:
Проверка диапазона заменена на более читаемую.

KAMKillJoy committed:
pixel переименован в pixel_format

KAMKillJoy committed:
фикс параметров экземпляра Commander

KAMKillJoy committed:
Кодеку vp9 добавлен контейнер mp4

KAMKillJoy committed:
Добавление опции аудио битрейта. Замена аудиокодека на opus для всех видеокодеков

KAMKillJoy committed:
Добавлен ридми

KAMKillJoy committed:
Убрана тавтология

KAMKillJoy committed:
Чуть исправлен стиль

KAMKillJoy committed:
добавлен комментарий к output_dir

KAMKillJoy committed:
новая хотелка

KAMKillJoy committed:
реформат pep8

KAMKillJoy committed:
Добавлено изменение заголовка терминала.

KAMKillJoy committed:
добавлен скрин меню для ридми

KAMKillJoy committed:
Update README.MD

добавил картинку

KAMKillJoy committed:
удалил отступ в пустой строке

KAMKillJoy committed:
добавлена опция "don't change" для pixel_format

KAMKillJoy committed:
Добавлен метод обработки ввода фпс

KAMKillJoy committed:
Добавлены дополнительные проверки корректности ввода и изменена последовательность очистки вывода

KAMKillJoy committed:
убраны избыточные проверки

KAMKillJoy committed:
исправлено сообщение об ошибке

KAMKillJoy committed:
добавлена проверка scale != "don't change"

KAMKillJoy committed:
добавлены комментарии к параметрам, обрабатываемым собственными методами в Menu

KAMKillJoy committed:
уточнение комментариев

KAMKillJoy committed:
актуализация ридми

KAMKillJoy committed:
попытка сделать более явными параметры с уникальной обработкой

KAMKillJoy committed:
редактура ридми

KAMKillJoy committed:
добавил комментарии ко всем кодекам, не смотря на дублирование

KAMKillJoy committed:
снова изменил значения параметров с уникальной обработкой

KAMKillJoy committed:
пустая строка в конце

KAMKillJoy committed:
Merge remote-tracking branch 'origin/main' into dev
# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
# the commit.

Perevedko commented on runner.py:
```diff
@@ -0,0 +1,88 @@
+        print(f"{i}. {c.name}")
+    while True:
+        choice = input("> ").strip()
+        try:
```
не совсем понятно, зачем тут try/except, если уже проверяешь на L20 на то, что такой индекс существует

KAMKillJoy replied:
Если юзер введёт набор букв `idx = int(choice)-1` выбросит ValueError

Perevedko replied:
принято, try/catch нужен для безопасного каста к Инту

Perevedko commented on runner.py:
```diff
@@ -0,0 +1,88 @@
+            idx = int(choice) - 1
+            if 0 <= idx < len(supported):
+                return supported[idx]
+        except ValueError:
```
почему этот эксепшон? exception ValueError
Raised when an operation or function receives an argument that has the right type but an inappropriate value, and the situation is not described by a more precise exception such as [IndexError](https://docs.python.org/3/library/exceptions.html#IndexError).

KAMKillJoy replied:
Если юзер введёт набор букв `idx = int(choice)-1` выбросит ValueError

Perevedko replied:
принято, try/catch нужен для безопасного каста к Инту

Perevedko commented on README.MD:
```diff
@@ -0,0 +1,73 @@
+
+**runner_%codec_name%** Раннер, который пропускает выбор кодека.
+
+**defaults.json** Стандартные настройки параметров кодека, которые применяются при запуске. Если этот файл удалить, то он будет создан заново. Тут можно поменять стандартные значения для своих типичных сценариев.
```
сомнительно. у тебя данные дублируются 2 раза, и в этом файле и в коде. можно сделать 2 файла, типа, defaults и options, и соответственно, псевдокод: if options.json not exist then `cp defaults.json options.json`. а `options.json` можно засунуть в .gitignore

Perevedko commented on commander.py:
```diff
@@ -0,0 +1,76 @@
+        self.codec = codec
+        self.settings = settings
+
+    def build_ffmpeg_command(self, file: str, output_dir) -> str:
```
очень большой метод получился, его бы декомпозировать

Perevedko commented on helpers.py:
```diff
@@ -0,0 +1,120 @@
+    print(f"Created defaults.json at {json_path}")
+
+
+import my_codecs
```
по pep-8 все импорты должны быть наверху файла

Perevedko commented on internal/menu.py:
```diff
@@ -0,0 +1,215 @@
+        self.codec = codec
+        self.settings = settings
+
+    def show(self):
```
я так понимаю что все методы в этом классе кроме show должны быть private

Perevedko commented:
> художника каждый может обидеть

Сапожника! К тебе бесплатно сеньор пришел ревюить код, тут радоваться надо! Основные моменты вроде отписал, в целом, если оно работает и ты для себя таким образом решил какие-то задачи, то это очень круто и ты молодец. А я ревьюю не тебя, а твой код.

Ещё моменты:
1. Декомпозиция методов в целом, есть, куда поработать тут
2. Видел класс, где есть и static и instance методы, не очень понятно, зачем так
3. У тебя внешняя зависимость в виде ffmpeg, это тоже надо учитывать. Ты тестировал на какой-то конкретной версии ffmpeg, нет гарантий, что api ffmpeg не изменится в следующих версиях. При старте можно проверять версию ffmpeg и писать, мол "разработано для такой-то, а у вас такая-то, нет гарантий что будет работать"

KAMKillJoy commented:
Я не тестил, потому пока не хотел сливать в main. Спасибо за ревью, сеньор помидор, поправлю.

KAMKillJoy committed:
импорты под пеп8

KAMKillJoy committed:
настройки перенесены в options.json

KAMKillJoy committed:
Сделал методы приватными

KAMKillJoy committed:
статик метод clear_screen() теперь вызывается как статик

KAMKillJoy committed:
декомпозировал build_ffmpeg_command()

KAMKillJoy committed:
добавил типизацию методам

KAMKillJoy committed:
Fix ожидаемого аргумента passes == "Two_Pass на passes == "Two-Pass

KAMKillJoy committed:
метод меню выбора кодека вынесен в menu.Meny

KAMKillJoy committed:
Добавление пропуска меню через runner и поддержка аргументов запуска

- теперь раннер может получать аргумент skip_menu в функцию main
- теперь можно запускать раннер с аргументами запуска

KAMKillJoy committed:
Актуализация ридми

KAMKillJoy committed:
фикс ридми

KAMKillJoy committed:
актуализация ридми

KAMKillJoy committed:
рефакторинг helpers

Perevedko commented on internal/commander.py:
```diff
@@ -0,0 +1,81 @@
+        if scale and scale != "don't change":
+            vf_list.append(f"scale={scale}")
+
+        if scale_fix:
```
абсурдно. т.к. дальше уже проверяешь на конкретные значения. переделай на if scale_fix == "pad" / elif scale_fix == "crop"

KAMKillJoy replied:
if на elif переделал, но думаю что проверку if scale_fix нужно оставить? Одно из значений, присваиваемое через GUI это None, так что мы можем сразу скипнуть этот блок.

Perevedko replied:
да ну, бред, мало выигрываешь от этого. лучше код сделать чище

KAMKillJoy replied:
Имхо это удобно выделяет логику scale_fix в отдельный блок. Хз, не могу себя заставить убрать.

KAMKillJoy replied:
Ладно, убрал.

Perevedko commented on commander.py:
```diff
@@ -0,0 +1,81 @@
+        if pixel_format and pixel_format != "don't change":
+            vf_list.append(f"format={pixel_format}")
+
+        vf = "-vf \"" + ",".join(vf_list) + "\"" if vf_list else ""
```
перепиши на single-quoted f-string

Perevedko commented on commander.py:
```diff
@@ -0,0 +1,81 @@
+
+
+class Commander:
+    def __init__(self, codec, settings: dict):
```
почему у codec нет аннотации типа?

Perevedko commented on internal/helpers.py:
```diff
@@ -0,0 +1,139 @@
+    if os.path.exists(defaults_path):
+        return  # Файл уже есть, ничего не делаем
+
+    defaults = {
```
всё ещё повторяется в двух местах в проекте. DRY

KAMKillJoy replied:
Убрал вообще defaults.json. Теперь есть options.json и его хардкодная стандартная версия, которая создаётся при отсутствии файла. Защита от дурака + возможность сбросить options.json если юзер что-то там навертел нерабочее.

Perevedko commented on helpers.py:
```diff
@@ -0,0 +1,139 @@
+        return "crop=trunc(iw/2)*2:trunc(ih/2)*2"
+
+    @staticmethod
+    def validate_resolution(width: int, height: int, even: bool = True) -> bool:
```
не вижу использований этого метода, он, вообще, нужен?

KAMKillJoy replied:
Нет, на данный момент оба метода не используются. Могут пригодиться в дальнейшем, но вряд ли. Пусть лежат? Да, можно было бы избавиться от класса вообще, но зачем?
Переписал validate_resolution() немного

Perevedko replied:
принцип YAGNI. убирай лишние методы

KAMKillJoy replied:
Убано

Perevedko commented on helpers.py:
```diff
@@ -0,0 +1,139 @@
+
+class ResolutionFixer:
+    @staticmethod
+    def pad():
```
зачем pad и crop это методы? это могли бы быть константы

Perevedko commented on helpers.py:
```diff
@@ -0,0 +1,139 @@
+        raise FileNotFoundError(f"{json_path} not found")
+
+    with open(json_path, "r", encoding="utf-8") as f:
+        all_defaults = json.load(f)
```
return прям здесь можно делать

Perevedko commented on converter.py:
```diff
@@ -0,0 +1,107 @@
+        return'''
+
+    # Выбор кодека
+    if preselected_codec:
```
вынеси все if/elif/else про кодеки в отдельный метод

Perevedko commented on converter.py:
```diff
@@ -0,0 +1,107 @@
+    print(f"All done! Total processing time: {total_duration}")
+
+    input("Press Enter to open the output folder, or close this window manually.")
+    if os.name == "nt":
```
можешь пооупражняться в ООП: сделать child classes для разных платформ и там уже будет платформо-специфичное поведение в методах

KAMKillJoy replied:
Сделал

Perevedko commented on helpers.py:
```diff
@@ -0,0 +1,139 @@
+        create_options_json()
+
+    with open(options_path, "r", encoding="utf-8") as f:
+        all_defaults = json.load(f)
```
аналогично, можно прям в этом блоке делать return

Perevedko commented on internal/commander.py:
```diff
@@ -0,0 +1,81 @@
+        vcodec = self.codec.vcodec
+        acodec = self.codec.acodec
+
+        if passes == "One-Pass":
```
тоже бы extract to method

KAMKillJoy replied:
Зачем? Да, метод очень грозодкий, но мне кажется что проблема не решится если мы его куски ещё и по углам разсуём. Переиспользования здесь не предвидется, потому мне кажется что читаемость только пострадает, а поддержка не выиграет ничего.

Perevedko replied:
читать станет проще намного метод. я бы рекомендовал тут, в #build_ffmpeg_command построить dict со значениями, который будешь передавать в методы __one_pass_cmd и __two_pass_cmd. сами строки 'One-Pass' и 'Two-Pass' нужно вынести в константы и использовать эти везде в проекте, где эти строки используются. это best practice и могу объяснить, почему

Perevedko commented on internal/menu.py:
```diff
@@ -0,0 +1,230 @@
+import os
+
+
+class Menu:
```
плохо, что тут всё смешано, и логика выбора, и логика представления. почитай про MVC паттерн, M у тебя, по сути, нет, а вот V и C есть, можно отделить

Perevedko commented on internal/menu.py:
```diff
@@ -0,0 +1,230 @@
+                'Input FPS (empty to go back, "r" to reset):').strip()
+            if val == "":
+                break
+            elif val.lower() == "r":
```
очень много магических литералов в этом файле

Perevedko commented:
в целом: нет смысла в корне держать все файлы, сделай так, чтоб в корне были только файлы, которые пользователь будет вызывать, а остальные положи в папку/папки. добавь shebangs файлам, которые пользователь будет непосредственно вызывать

KAMKillJoy committed:
Удалён ненужный метод

KAMKillJoy committed:
методы pad() и crop() заменены на константы

KAMKillJoy committed:
замена сборки строки vf на более читаемую

KAMKillJoy committed:
Добавлена аннотация codec: Codec

KAMKillJoy committed:
Рефакторинг validate_resolution()

KAMKillJoy committed:
defaults.json ликвидирован

KAMKillJoy committed:
убраны лишние методы

KAMKillJoy committed:
реформат return

Perevedko commented on internal/helpers.py:
```diff
@@ -0,0 +1,123 @@
+    Возвращает список объектов Codec, доступных в my_codecs.
+    """
+    return [
+        obj for name, obj in vars(my_codecs).items()
```
vars это моветон

Perevedko replied:
лучше предопределить такой список. как вариант, при создании объекта, можно записывать ссылку на него в какой-нибудь список. но лучше, наверное, явно определять список кодеков

KAMKillJoy replied:
Реализовал через список, куда экземпляр добавляется при создании.

KAMKillJoy commented:
TO DO:
- [x] runner -> smth

KAMKillJoy committed:
Перенос внутренних модулей в отдельную директорию

KAMKillJoy committed:
актулизация импортов

KAMKillJoy committed:
ремофрмат обработки видеофильтров

KAMKillJoy committed:
runner переименован в converter

KAMKillJoy committed:
починил импорты

ide не починила ипорты при переносе модулей

KAMKillJoy committed:
Update README.MD

удалил неактуальные строки про defaults.json

KAMKillJoy committed:
фикс ридми

KAMKillJoy committed:
реализация универсального интерфейса для ОС-специфичных взаимодействий

PlatformAdapter ->  WindowsAdapter(PlatformAdapter) и PosixAdapter(PlatformAdapter) в helpers.py

KAMKillJoy committed:
shebang и небольшая коррекция порядка импортов

Perevedko commented on README.MD:
```diff
@@ -0,0 +1,130 @@
+После перебора весьма непродуктивных предложений, удалось таки понять какой вариант реализации был бы наиболее логичным
+и удобным для меня. Его вы видите тут.
+
+## Суть реализации
```
это лучше вынести непосредственно в файлы/в сущности. предполагаемому пользователю вряд ли интересно это, а программисту будет неудобно ориентироваться

Perevedko commented on README.MD:
```diff
@@ -0,0 +1,130 @@
+- погуглить как включить Drag & Drop файлов на файлы Python.
+- добавить скрипт в виде кнопки в вашем файловом менеджере а-ля Total Commander.
+
+### Аргументы запуска
```
это, наверное, лучше всего вынести наверх доки

Perevedko commented on converter.py:
```diff
@@ -0,0 +1,103 @@
+    codecs = supported_codecs()
+
+    files = args.files
+    '''if not files:
```
зачем тут строка? это коммент?

KAMKillJoy replied:
это закоменченый код. для отладки мне удобно запускать без передачи файлов скрипту

Perevedko commented on internal/helpers.py:
```diff
@@ -0,0 +1,122 @@
+        os.system("clear")
+
+
+if os.name == "nt":
```
ставлю лайк

Perevedko commented on internal/helpers.py:
```diff
@@ -0,0 +1,122 @@
+    ]
+
+
+class PlatformAdapter:
```
я не уверен, что это каноничный способ реализовать такую идею. он достаточный, но каноничный в пайтоне, ЕМНИП6 это abc.ABC. в будущем рекомендую изучить эти механизмы, abc.ABC и abc. abstractmethod. сейчас это не нужно править

Perevedko commented on internal/commander.py:
```diff
@@ -0,0 +1,85 @@
+                f'-c:a {acodec} -b:a {audio_bitrate}k "{output_file}_2pass.{container}"'
+            )
+        else:
+            raise ValueError(f"Passes should be one of 'One-Pass', 'Two-Pass'")
```
константы вынесенные надо будет использовать и здесь

Perevedko commented on internal/commander.py:
```diff
@@ -0,0 +1,85 @@
+                f'-preset {str(preset)} -crf {str(crf)} -pass 2 '
+                f'-c:a {acodec} -b:a {audio_bitrate}k "{output_file}_2pass.{container}"'
+            )
+        else:
```
ещё один best practice это early fail. такой рейт надо делать в начале метода. да и, вообще, посмотри в своём проекте, может быть где-то ещё не хватает early fail.

"Fail Fast" and "Fail Early" are related software design and agile development principles that advocate for error detection and handling as soon as possible to prevent incorrect states from persisting and causing more significant problems later

Perevedko commented on internal/helpers.py:
```diff
@@ -0,0 +1,122 @@
+    """
+    Создаёт options_json с базовыми значениями кодеков
+    """
+    defaults = {
```
удали и заменеджи гитом это. например os.system('git checkout -f options.json')

Perevedko commented on internal/menu.py:
```diff
@@ -0,0 +1,230 @@
+                except (ValueError, IndexError):
+                    print("Invalid choice, try again.")
+
+    def __handle_scale(self):
```
тоже надо декомпозировать

Perevedko commented on internal/menu.py:
```diff
@@ -0,0 +1,230 @@
+                self.settings["fps"] = "don't change"
+                break
+            try:
+                val_int = int(val)
```
лишняя переменная

Perevedko commented on internal/my_codecs.py:
```diff
@@ -0,0 +1,88 @@
+class Codec:
+    def __init__(
+            self,
+            name: str,
```
дай всем параметрам дефолтное значение None, так при вызовах обязательно нужно будет явно указывать keyword arguments. а щас первые 2 можно передать без названий

Perevedko commented on internal/commander.py:
```diff
@@ -0,0 +1,85 @@
+
+        # обработка scale_fix
+        if scale_fix == "pad":
+            scale_fix_filter = helpers.ResolutionFixer.PAD
```
лишняя переменная. по названию константы должно быть понимание, что это

Perevedko commented on internal/commander.py:
```diff
@@ -0,0 +1,85 @@
+            vf_list.append(scale_fix_filter)
+
+        elif scale_fix == "crop":
+            scale_fix_filter = helpers.ResolutionFixer.CROP
```
лишняя переменная. по названию константы должно быть понимание, что это

Perevedko commented on internal/commander.py:
```diff
@@ -0,0 +1,85 @@
+            vf_list.append(scale_fix_filter)
+
+        # обработка  fps
+        if fps and fps != "don't change":
```
magic literal (string)

Perevedko commented on internal/commander.py:
```diff
@@ -0,0 +1,85 @@
+        scale_fix = self.settings.get("scale_fix")
+
+        # обработка scale
+        if scale and scale != "don't change":
```
magic literal (string)

Perevedko commented on internal/commander.py:
```diff
@@ -0,0 +1,85 @@
+            vf_list.append(f"scale={scale}")
+
+        # обработка scale_fix
+        if scale_fix == "pad":
```
magic literal (string)

Perevedko commented on internal/commander.py:
```diff
@@ -0,0 +1,85 @@
+        if scale and scale != "don't change":
+            vf_list.append(f"scale={scale}")
+
+        # обработка scale_fix
```
удали коммент, он лишний

KAMKillJoy committed:
убраны лишние переменные

KAMKillJoy committed:
литерал "don't change" вынесен в константу

KAMKillJoy committed:
убрана лишняя переменная

KAMKillJoy committed:
добавлен коммент к константе

KAMKillJoy committed:
Форсирована передача параметров класса Codec в виде keyarg

Perevedko commented:
фичреквест: при старте проверять наличие ffmpeg в системе и сразу крашиться, если он не обнаружен
т.е. прикинь, пользователь запустил, натыкал всё, а ему программа такая ХОБА ПРИВЕТ в конце flow уже

upd: сделано

KAMKillJoy committed:
Реализована функция проверки наличия ffmpeg

KAMKillJoy committed:
Изменение импортов

Импорт из helpers отдельных сущностей достиг предела, теперь ипортируется весь модуль.

KAMKillJoy committed:
Выбор кодека вынесен в отдельный метод

KAMKillJoy committed:
актуализация ридми

KAMKillJoy committed:
улучшение читаемости логических поверок

KAMKillJoy committed:
Изменена опция пресета vp9 на актуальную

KAMKillJoy commented:
И вот на этом славном моменте я понял что меню у меня составляется "автоматически", исходя из экземпляра Codec, а вот строка собирает предопределённые аргументы. 
Суть: vp9 вместо preset имеет аналогичный аргумент deadline и мы можем добавить экземпляру vp9 такой параметр и он отразится в меню. Только вот commander ждёт именно preset и в итоговой команде захардкожен -preset. Это ломает всю базу реализации об колено, но это должно было произойти.

KAMKillJoy committed:
переработка vp9 под более универсальную обработку меню

KAMKillJoy committed:
Актуализация вывода пунктов меню

KAMKillJoy committed:
Добавлена константа с текстом для resetteble параметров

KAMKillJoy committed:
исправлен тип параметра фпс

KAMKillJoy committed:
редактура текста

KAMKillJoy committed:
Изменение логики меню под новый формат Codec

KAMKillJoy committed:
добавлена проверка допустимых значений

KAMKillJoy committed:
Актуализация Codec

- скорректированы значения flag
- скорректированы значения context
- актуализированны экземпляры svt-av1 и hevc

KAMKillJoy committed:
убран дебаг-принт

KAMKillJoy committed:
актуализация commander под data driven codec.py

KAMKillJoy committed:
FIRST_PASS_SKIP_PARAMS

содержит парметры, которые надо игнорировать в первом прогоне в TWO PASS

KAMKillJoy committed:
Убрано 2pass для svt-av1 и добавлена обработка None в settings.passes

KAMKillJoy commented:
> И вот на этом славном моменте я понял что меню у меня составляется "автоматически", исходя из экземпляра Codec, а вот строка собирает предопределённые аргументы. Суть: vp9 вместо preset имеет аналогичный аргумент deadline и мы можем добавить экземпляру vp9 такой параметр и он отразится в меню. Только вот commander ждёт именно preset и в итоговой команде захардкожен -preset. Это ломает всю базу реализации об колено, но это должно было произойти.

Исправлено. Получилась годная data driven штучка. Ну или не годная, но я считаю что годная.

KAMKillJoy committed:
Актуализация стандартных настроек

KAMKillJoy committed:
Уточнены единицы измерения аудиобитрейта

KAMKillJoy committed:
Оптимизация импортов и немного типизации

KAMKillJoy committed:
Изменение пунктов меню

KAMKillJoy committed:
Вернул потерянный фикс нечётного разрешения

KAMKillJoy committed:
актуализация стандартных настроек и вынесение литерала сброса опции

KAMKillJoy committed:
фикс разрешения вынесен внутрь проверки

KAMKillJoy committed:
перенёс фикс разрешения в Menu

KAMKillJoy committed:
актуализация подсказки старта рендера

KAMKillJoy commented:
to do:
в основных меню инпут отображает "> ", а в подменю нет. Надо подумоть как сделать.

KAMKillJoy committed:
убрал разделение опций запятыми

KAMKillJoy committed:
убрал ненужные кавычки в команде

KAMKillJoy committed:
добавил флаг для scale

KAMKillJoy committed:
подобрал полее адекватные стандартыне настройки

KAMKillJoy committed:
косметические изменения запроса инпута

KAMKillJoy committed:
Актуализация подсказок gui и ридми

KAMKillJoy commented:
> to do: в основных меню инпут отображает "> ", а в подменю нет. Надо подумоть как сделать.

Сделано. везде "> "

KAMKillJoy committed:
починил составление подстроки опций

KAMKillJoy committed:
починил составление подстроки опций ещё

KAMKillJoy committed:
svt-av1 теперь требует чётных сторон

KAMKillJoy committed:
изменение сброки подстрок фильтров. исправление игнорирования опции scalefix

KAMKillJoy committed:
Изменил логику составления списка поддерживаемых кодеков

KAMKillJoy committed:
Реализация абстрактного класса теперь более питонистая. Логика та же.

KAMKillJoy committed:
Отказался от ручной обработки нечётных значений разрешения

Раньше был отдельный метод. Теперь подменю прописано в my_codecs что гораздо логичнее

KAMKillJoy committed:
fix attempt

KAMKillJoy committed:
Удаление лишних пробелов в строке фильтров

KAMKillJoy committed:
добавил батник для drag and drop

KAMKillJoy committed:
оказывается я потерял указание собственно видеокодека

KAMKillJoy committed:
добавление tune для svt-av1

KAMKillJoy committed:
обновление to do в ридми

KAMKillJoy committed:
починил two pass

KAMKillJoy committed:
добавил tune в vp9

KAMKillJoy committed:
актуализация to do

KAMKillJoy committed:
добавлен tune для hevc

KAMKillJoy committed:
изменил лейблы опций tune

KAMKillJoy committed:
Восстановил data driven концепцию + реф

1. ввёл новые контекст для уникальных параметров кодека, которые ффмпег сам не умеет пробрасывать из глобальных параметров
2. зарефакторил сборку итоговой команды так чтобы не было лишних пробелов

KAMKillJoy committed:
обновление gitignore

KAMKillJoy committed:
Рефакторинг контекстов параметров + новый параметр scale filter

KAMKillJoy committed:
восстановил data driven

теперь есть новый контекст "%some param% flag". при обработке каждого параметра в настройках ищет наличие флагов этого параметра. никогда не пригодится, но ведь работает!

KAMKillJoy committed:
добавил универсальности (катомные флаги параметров фильтров) и немного исправил нейминг

KAMKillJoy committed:
потужный рефакторинг

- вынес литералы из my_codecs в codec_schema
- вынес проверку на опцию, которую не надо применять, в метод is_skip_param(value)
- поченил сброс некоторых параметрой в menu

KAMKillJoy committed:
проект переименован

KAMKillJoy committed:
починил меню

забыл прописать константы вместо литералов

KAMKillJoy committed:
поченил обращение к enum

на практике оно работало из-за странного магического поведения enum в питоне

KAMKillJoy committed:
починил парсинг фильтров

я всё ещё его ненавижу и должен накорю переделать

KAMKillJoy committed:
раскоментил проверку на отсутствие файлов

KAMKillJoy committed:
Класс Type переименован в ParamType + pep8
